### DISCUSSION TITLE
Filter function does not have access to global/outer scope

### DISCUSSION THREAD

**Discussion by jon-v2** [vincjo/svelte-simple-datatables#70] [Q&A]
From my understanding, you can set the data-key property to equal a function for custom filtering. This is great except you cant use a predefined function within your component i.e:

`const convertDate = (date) => { return moment(date).utc().format("MMM Do YYYY"); }`

...

`<th data-key={(row) => convertDate(row.date)}>Date</th>`

this will throw an exception saying convertDate does not exist. Please assist.

Edit: that being said, you can't reference "moment" directly either (as it will throw an exception saying moment doesn't exist)

**Comment by kevin4dhd** [comment]
any solution?
